api/Updated-Metrics.md

```diff
@@ -0,0 +1,394 @@
+# Updates for Metrics
+
+This is an update for the existing metrics document, which is being left in place for now as a point of comparison.
+
+## Assumed data
+
+In the following we assume that we have variables of the following form defined:
+
+```python
+y_true = [0, 1, 0, 0, 1, 1, ...]
+y_pred = [1, 0, 0, 1, 0, 1, ...] # Content can be different for other metrics (see below)
+A_1 = [ 'C', 'B', 'B', 'C', ...]
+A_2 = [ 'M', 'N', 'N', 'P', ...]
+A = pd.DataFrame(np.transpose([A_1, A_2]), columns=['SF 1', 'SF 2'])
+
+weights = [ 1, 2, 3, 2, 2, 1, ...]
+```
+
+We actually seek to be very agnostic as to the contents of the `y_true` and `y_pred` arrays; the meaning is imposed on them by the underlying metrics.
+Here we have shown binary values for a simple classification problem, but they could be floating point values from a regression, or even collections of classes and associated probabilities.
+
+## Basic Calls
```

Comment:
I think that before we discuss the details below, it would be good to give examples of how one would go about using the current API versus the new API to get some specific things done. I don't quite see that here. What I'm looking for is something like "test cases" [here](https://scikit-learn-enhancement-proposals.readthedocs.io/en/latest/slep006/proposal.html#test-case-setup). Maybe we should consider the following test cases / tasks:

1. Run cross validation with a demographic parity (as a fairness metric) and balanced error rate (as the performance metric) when our goal is to optimize balanced error rate while demographic parity difference is at most 0.02. We should perhaps review this in the context of [Adrin's PR in sklearn](https://github.com/scikit-learn/scikit-learn/pull/16079)?
1. Report one disaggregated metric in a data frame.
1. Report several disaggregated metrics in a data frame.
1. Report several performance and fairness metrics of several models in a data frame.
1. Report several performance and fairness metrics as well as some disaggregated metrics of several models in a data frame.

Reply:
For (1), please can you let me know what you feel is missing from the block starting on line 59. Similarly for (2) and line 272.

Can you give some examples of what you'd like to see for (3) and (4). If you mean combining metrics computed via `group_summary()` (or whatever our ultimate solution is) and other means, I'm not sure that falls within the remit of Fairlearn.